test/AutoDiff/SILOptimizer/differentiation_diagnostics.swift

```diff
@@ -741,7 +741,7 @@ struct TF_675 : Differentiable {
 let _: @differentiable(reverse) (Float) -> Float = TF_675().method
 
 // TF-918: Test parameter subset thunk + partially-applied original function.
-let _: @differentiable(reverse) (Float, Float) -> Float = (+) as @differentiable(reverse) (Float, @noDerivative Float) -> Float
+let _: @differentiable(reverse) (Float, @noDerivative Float) -> Float = (+) as @differentiable(reverse) (Float, @noDerivative Float) -> Float
```

Comment:
I'm surprised this test was there!  Maybe it was meant to test function conversions to a function with more `@noDerivative`. That should be possible with "subset parameter thunks".

```swift
let _: @differentiable(reverse) (Float, @noDerivative Float) -> Float = (+) as @differentiable(reverse) (Float, Float) -> Float
```

Is the above possible today?

Reply:
Yes, the above example is also possible. I think it is the same as `test1` in the new LIT test, but we can add it to `differentiation_diagnostics.swift` as well.

Reply:
Does it actually work in a runtime test?